pandas/core/apply.py

```diff
@@ -1824,12 +1824,15 @@ def relabel_result(
             fun = [
                 com.get_callable_name(f) if not isinstance(f, str) else f for f in fun
             ]
-            col_idx_order = Index(s.index).get_indexer(fun)
-            s = s.iloc[col_idx_order]
+            col_idx_order = list(Index(s.index).get_indexer(fun))
+            col_idx_order = [i for i in col_idx_order if 0 <= i < len(s)]
+            if col_idx_order:
+                s = s.iloc[col_idx_order]
```

Comment:
Instead I think you can filter `col_idx_order` where it's equal to `-1`. See the `get_indexer` docstring